app/views/questionnaire.py

```diff
@@ -460,10 +483,9 @@ def update_questionnaire_store_with_answer_data(questionnaire_store, location, a
         questionnaire_store.completed_blocks.append(location)
 
 
-def _check_same_survey(eq_id, form_type, collection_id):
-    metadata = get_metadata(current_user)
-    current_survey = eq_id + form_type + collection_id
-    metadata_survey = metadata['eq_id'] + metadata['form_type'] + metadata['collection_exercise_sid']
+def _check_same_survey(current_eq_id, current_form_type, current_collection_id, stored_eq_id, stored_form_type, stored_collection_id):
```

Comment:
I think it would be clearer if we named the values with url instead of current e.g.`url_eq_id` and session instead of stored e.g. `session_eq_id`